gene2phenotype_project/gene2phenotype_app/serializers.py

```diff
@@ -1205,6 +1819,134 @@ def update(self, instance, validated_data):
 
         return instance
 
+    @transaction.atomic
+    def publish(self, data):
```

Comment:
No, I think the updating needs to be done, because the updated data needs to reflect on the DB before it is published, I may be wrong though @seeta-ramaraju

Reply:
The frontend can save it, before publishing.

Reply:
Yeah but the frontend can not save it to the DB right? I am saying for example if there is a draft and the person adds more information or changes something in the already existing structure then clicks publish, at the backend, the process should be update->publish->delete from curation table 
Otherwise it throws an error.

Reply:
I also think the process should always be update->publish->delete from curation table, no matter if the updated or not. Just to make sure we are capturing the latest json_data being sent in the request from frontend.

Reply:
Yeah you are right, would be better from the frontend.